File: common/src/main/java/com/sumian/common/social/OpenEngine.java
package com.sumian.common.social;

import android.app.Activity;
import android.content.Context;
import androidx.annotation.DrawableRes;

import com.sumian.common.social.analytics.OpenAnalytics;
import com.sumian.common.social.login.OpenLogin;
import com.umeng.analytics.MobclickAgent;
import com.umeng.commonsdk.UMConfigure;
import com.umeng.socialize.ShareAction;
import com.umeng.socialize.UMShareListener;
import com.umeng.socialize.bean.SHARE_MEDIA;
import com.umeng.socialize.media.UMImage;
import com.umeng.socialize.media.UMWeb;


/**
 * Created by jzz
 * on 2017/12/27.
 * desc:
 */

public class OpenEngine {

    private OpenLogin mOpenLogin;
    private OpenAnalytics mOpenAnalytics;

    /**
     * @param context    上下文，不能为空
     * @param appKey     【友盟+】 AppKey
     * @param channel    【友盟+】 Channel
     * @param pushSecret Push推送业务的secret，需要集成Push功能时必须传入Push的secret，否则传空。
     */
    public static void init(Context context, boolean debug, String appKey, String channel, String pushSecret) {
        UMConfigure.init(context,
                appKey,
                channel,
                UMConfigure.DEVICE_TYPE_PHONE,
                pushSecret);
        UMConfigure.setLogEnabled(debug);
        // 开启日志上报
        MobclickAgent.setCatchUncaughtExceptions(true);
    }

    public OpenEngine create(Context context, boolean isDebug, String appId, String appSecret) {
        this.mOpenLogin = new OpenLogin().init(context, appId, appSecret);
        this.mOpenAnalytics = new OpenAnalytics().init(context, isDebug);
        return this;
    }

    public OpenLogin getOpenLogin() {
        return mOpenLogin;
    }

    public OpenAnalytics getOpenAnalytics() {
        return mOpenAnalytics;
    }

    public void shareWeb(Activity activity, String url, String title, String description, String thumb, SHARE_MEDIA shareMedia) {
        UMWeb web = new UMWeb(url);
        web.setTitle(title);
        web.setDescription(description);
        web.setThumb(new UMImage(activity, thumb));
        new ShareAction(activity)
                .withMedia(web)
                .setPlatform(shareMedia)
                .share();
    }

    public void shareWebForCallback(Activity activity, String url, String title, String description, @DrawableRes int thumb, SHARE_MEDIA shareMedia, UMShareListener umShareListener) {
        UMWeb web = new UMWeb(url);
        web.setTitle(title);
        web.setDescription(description);
        web.setThumb(new UMImage(activity, thumb));
        new ShareAction(activity)
                .withMedia(web)
                .setCallback(umShareListener)
                .setPlatform(shareMedia)
                .share();
    }
}

File: common/src/main/java/com/sumian/common/social/OpenEngine.kt
package com.sumian.common.social

import android.app.Activity
import android.content.Context
import androidx.annotation.DrawableRes
import com.sumian.common.h5.bean.ShareData
import com.sumian.common.social.analytics.OpenAnalytics
import com.sumian.common.social.login.OpenLogin
import com.umeng.analytics.MobclickAgent
import com.umeng.commonsdk.UMConfigure
import com.umeng.socialize.ShareAction
import com.umeng.socialize.UMShareListener
import com.umeng.socialize.bean.SHARE_MEDIA
import com.umeng.socialize.media.UMImage
import com.umeng.socialize.media.UMWeb
import java.io.File


/**
 * Created by jzz
 * on 2017/12/27.
 * desc:
 */

class OpenEngine {

    var openLogin: OpenLogin? = null
        private set
    var openAnalytics: OpenAnalytics? = null
        private set

    fun create(context: Context, isDebug: Boolean, appId: String, appSecret: String): OpenEngine {
        this.openLogin = OpenLogin().init(context, appId, appSecret)
        this.openAnalytics = OpenAnalytics().init(context, isDebug)
        return this
    }

    fun shareUrl(activity: Activity, url: String, title: String, description: String, @DrawableRes thumb: Int, shareMedia: SHARE_MEDIA, umShareListener: UMShareListener?) {
        val web = UMWeb(url)
        web.title = title
        web.description = description
        web.setThumb(UMImage(activity, thumb))
        ShareAction(activity)
                .withMedia(web)
                .setCallback(umShareListener)
                .setPlatform(shareMedia)
                .share()
    }

    fun shareUrl(activity: Activity, url: String?, title: String?, description: String?, thumbUrl: String?, shareMedia: SHARE_MEDIA, umShareListener: UMShareListener? = null) {
        val web = UMWeb(url)
        web.title = title
        web.description = description
        web.setThumb(UMImage(activity, thumbUrl))
        ShareAction(activity)
                .withMedia(web)
                .setCallback(umShareListener)
                .setPlatform(shareMedia)
                .share()
    }

    fun shareImageFile(activity: Activity, file: File, text: String, shareMedia: SHARE_MEDIA, umShareListener: UMShareListener? = null) {
        val image = UMImage(activity, file)//本地文件
        image.setThumb(UMImage(activity, file))
        ShareAction(activity)
                .setPlatform(shareMedia)
                .withText(text)
                .withMedia(image)
                .setCallback(umShareListener)
                .share();
    }

    fun shareUrl(activity: Activity, shareData: ShareData, umShareListener: UMShareListener? = null) {
        shareUrl(activity, shareData.url, shareData.title, shareData.desc, shareData.icon, shareData.getPlatformEnum(), umShareListener)
    }

    companion object {

        /**
         * @param context    上下文，不能为空
         * @param appKey     【友盟+】 AppKey
         * @param channel    【友盟+】 Channel
         * @param pushSecret Push推送业务的secret，需要集成Push功能时必须传入Push的secret，否则传空。
         */
        fun init(context: Context, debug: Boolean, appKey: String, channel: String, pushSecret: String) {
            UMConfigure.init(context,
                    appKey,
                    channel,
                    UMConfigure.DEVICE_TYPE_PHONE,
                    pushSecret)
            UMConfigure.setLogEnabled(debug)
            // 开启日志上报
            MobclickAgent.setCatchUncaughtExceptions(true)
        }
    }
}
